src/test/feature-math.spec.ts

```diff
@@ -335,5 +338,21 @@ describe('feature-math', () => {
         expect(await contract.getCompare28(dictA, dictB)).toBe(true);
         expect(await contract.getCompare28(dictB, dictA)).toBe(true);
         expect(await contract.getCompare28(dictA, dictA)).toBe(false);
+
+        // Test advanced math operations
+        for (let num = 1n; num <= 100n; num++) {
+            expect(await contract.getLog2(num)).toBe(
+                BigInt(Math.floor(Math.log2(Number(num))))
+            );
+        }
+
+        for (let num = 1n; num <= 10n; num++) {
+            for (let base = 2n; base <= 10; base++) {
+                const a = Math.floor(Math.log2(Number(num)));
+                const b = Math.floor(Math.log2(Number(base)));
+                const c = BigInt(Math.floor(a / b));
```

Comment:
we want `c` to be something like `floor(log2(num) / log2(base))`, just like in the comment above